dashboard/src/pages/packages/index.vue

```diff
@@ -113,6 +128,14 @@ watch(
     return packageStore.fetchPackages(1);
   },
 );
+
+watch(
+  () => route.query.name,
+  (newSearch) => {
+    packageStore.filters.name = newSearch as string;
+    return packageStore.fetchPackages(1);
```

Comment:
When watching an input and making API requests based on its value, we should consider a few things:

- Debouncing/Throttling: making sure the request doesn't happen each keystroke.
- Request Cancellation: abort previous requests and prevent race conditions.
- Deduplication: about fetching the same query, caching results.

I'm not sure what would be the best way to implement these using our client and Vue but, ideally, we would create something that is reusable in other search inputs. **Doesn't need to happen in this PR.**

Also, it would be nice to provide some "Loading..." visual effect and we should consider pagination too, returning to the first page if the input changes.

Reply:
Because the API request is fired by submitting the form (well, in a Vue.js kind of SPA version of form submission) I think these issues are probably not as much of a concern.  I was originally using Debounced API requests, but it didn't seem necessary without the "onChange" requests.

`packageStore.fetchPackages(1);` does fetch the first page of the results (that's the 1 parameter).  The pager doesn't remember it's position when you click on a package on the list right now, but that's a problem to address in a different PR I think. :)

The loading is so fast right (the API response time was in the < 10 millisecond range when I tested) now I don't think a "Loading..." effect is necessary, but I think we could add it later if necessary.

Reply:
Oh, I see, this watches the route! Why do we need it then? If it's only changed from `doSearch()` it should be enough to fetch there.

Cool about pagination, didn't notice that 1. Loading speeds may change a lot in other environments not running everything locally, but yeah, we can add it later.

Reply:
There is lots I don't know about how Vue.js does routing, watchers and page flow so the way I've implemented this may not be ideal and I'm always open to other suggestions.  What I do know if I take the route watcher away, when I click the "back" button on the browser the URL in the browser bar updates appropriately, but the packages shown don't update to reflect the previous search value. :man_shrugging:

Reply:
I see that, I would expect that to trigger the `useAsyncState` again but it doesn't seem to do it. Leave it there then, thanks for noticing such details ;)

Reply:
Yeah, I would expect the the back button reload to trigger `useAsyncState` too, but I guess maybe Vue overrides the browser behaviour here to avoid reloading the whole app?